chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -1861,8 +1891,20 @@ impl PeerManagerActor {
             return RegisterPeerResponse::InvalidNonce(Box::new(last_edge.unwrap().clone()));
         }
 
-        if msg.other_edge_info.nonce >= Edge::next_nonce(last_nonce) + EDGE_NONCE_BUMP_ALLOWED {
-            debug!(target: "network", nonce = msg.other_edge_info.nonce, last_nonce, ?EDGE_NONCE_BUMP_ALLOWED, ?self.my_peer_id, ?msg.peer_info.id, "Too large nonce");
+        if let Ok(nonce) = i64::try_from(msg.other_edge_info.nonce) {
+            if let Some(nonce_timestamp) = Edge::nonce_to_utc(nonce) {
+                if (self.clock.now_utc() - nonce_timestamp).abs() < EDGE_NONCE_MAX_TIME_DELTA {
```

Comment:
Wait, no, this doesn't feel right -- nonces are user-controlled, so this can overflow, or at least it's not immediately obvious why this won't overflow. Could you add a test which uses `i64::MAX` as a nonce?

Reply:
good point - I've added a bunch of additoinal tests (both around i64MAX and around max date timestamps)